Subject: crypto: Remove duplicate srcu init
From: Milan Broz <mbroz@redhat.com>
Date: Tue, 30 Oct 2012 16:27:18 +0100

In peterz-srcu-crypto-chain.patch the blocking notifier is changed to
srcu notifier and added initialization to module init fucntion.
Later, in crypto-make-core-static-and-init-scru-early.patch, is that
initialization added also to core_initcall(), but not removed from
Peter's patch. So the initializer is called twice which can wipe out
already registered notifiers. This cause a failure in initialization
of larval algorithms, like e.g. cbc(aes).

Remove the old one.

Signed-off-by: Milan Broz <mbroz@redhat.com>
Cc: Tvrtko Ursulin <tvrtko@ursulin.net>
Cc: dm-crypt@saout.de
Cc: okozina@redhat.com
Cc: u.kleine-koenig@pengutronix.de
Link: http://lkml.kernel.org/r/508FF1D6.3030900@redhat.com
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>

---
 crypto/algapi.c |    1 -
 1 file changed, 1 deletion(-)

Index: linux-stable/crypto/algapi.c
===================================================================
--- linux-stable.orig/crypto/algapi.c
+++ linux-stable/crypto/algapi.c
@@ -956,7 +956,6 @@ EXPORT_SYMBOL_GPL(crypto_xor);
 
 static int __init crypto_algapi_init(void)
 {
-	srcu_init_notifier_head(&crypto_chain);
 	crypto_init_proc();
 	return 0;
 }
